### DISCUSSION TITLE
about javascript's Date Object, year 1582's lost 10 days problem and previous Date's leap years problem

### DISCUSSION THREAD

**Discussion by hh-in-zhuzhou** [cosinekitty/astronomy#284] [General]
I use this library to draw planet and sun, and paired with another calendar like Chinese lunar calendar and other ancient calendars，so I often use years before 1582，which let me found the errors in javascript's Date Object.
I found that javascript cannot support problem of Oct in year 1582, the Date Object can out put 1582/10/6 which is not exists。 so what about if this library process Dates before 1582 ?
welcome suggestions, thanks.

**Comment by cosinekitty** [comment]
I'm not sure what error you are encountering, but my guess is that it has something to do with the difference between the Julian calendar (before 1582) and the Gregorian calendar (after 1582). As far as I can tell, for consistency, JavaScript always uses the modern Gregorian calendar, even for years before 1582. If you want to support a different calendar like Julian, you would need to find some function to convert from Julian to Gregorian to pass into the JavaScript Date object.

**Reply by lulunac27a** [reply]
October 15, 1582, was the first day of the Gregorian Calendar, a day after the Julian day of October 4, 1582. This is because it aligns with the actual solar year, so the equinoxes and solstices align with the same date.
